client/sysinfo/src/sysinfo.rs

```diff
@@ -319,6 +325,44 @@ pub fn benchmark_disk_random_writes(directory: &Path) -> Result<u64, String> {
 
 	// We only wrote half of the bytes hence `SIZE / 2`.
 	benchmark("disk random write score", SIZE / 2, MAX_ITERATIONS, MAX_DURATION, run)
+		.map(|s| s as u64)
+}
+
+/// Verify signatures of different random data `reps` many times.
+pub fn benchmark_sr25519_verify(
+	reps: usize,
+	duration: Duration,
+	input_len: usize,
+) -> Result<f64, String> {
+	let pair = sr25519::Pair::from_string("//Alice", None).unwrap();
+
+	let mut rng = rng();
+	let mut msgs = Vec::new();
+	let mut sigs = Vec::new();
+
+	for _ in 0..reps {
+		let mut msg = vec![0u8; input_len];
+		rng.fill_bytes(&mut msg[..]);
+
+		sigs.push(pair.sign(&msg));
+		msgs.push(msg);
+	}
+
+	let run = || {
+		for (sig, msg) in sigs.iter().zip(msgs.iter()) {
+			let mut ok = sr25519_verify(&sig, &msg[..], &pair.public());
+			// Clobber the result.
+			unsafe {
+				let mut raw = std::slice::from_raw_parts_mut(
+					&mut ok as *mut bool as *mut u8,
+					core::mem::size_of::<bool>(),
+				);
+				clobber(&mut raw);
+			}
+		}
+		Ok(())
+	};
+	benchmark("sr25519 verification score", reps * input_len, 1, duration, run)
```

Comment:
Since you passed `1` as the maximum number of iterations the `duration` doesn't actually do anything. (:

The `benchmark` function stops either when we hit the given number of iterations, or when we hit a given duration, where the duration is only checked after each iteration.

So basically this should be adjusted so that:

1) One iteration does enough work that compared to the time it takes to run it calling `std::time::Instant::now()` is negligible. Basically pick some standard values of `reps` and `input_len` for the benchmark that on modern hardware a single iteration runs not too short and not long. (Less than 100ms and more than 10ms would probably be good? If you turn on the `trace` logs you'll see a line for how long the benchmark took and how many iterations it managed to do in that time.)
2) Parametrize the benchmark not by `reps` and `input_len` but by `max_iterations` and `max_duration`. The `max_iterations` *is* effectively what `reps` is supposed to be, and it makes sense to standardize on a concrete `input_len` because that will most likely change what results you'll get - that is, for different values of `input_len` you'll most likely get different throughputs, so those would effectively be slightly different benchmarks.

Then when calling this function from the CLI subcommand you'll have a choice - you can either set `max_durations` to `MAX` and have it bound only by `duration`, you can set `duration` to a large value and have it bound only by the number of iterations, or you can set both. (For the subcommand I guess bounding it only by time makes the most sense to me?)